fit_dict functionality, removed ACE dependency

casv2 opened the pull request:
I think this solves the problems we had. I'm trying to test this now from a clean install. Not sure why I get an error trying to pull the `/dev` branch `add IPFitting#dev` gives me `ERROR: invalid git HEAD (reference 'refs/heads/master' not found)`. Pulling `/main` works (`add ACE IPFitting`) so once this gets merged I think we should be OK.

I'll add a tutorial on how to fit a simple ACE using the new interface and how to use it in Python ASE as well as LAMMPs.

casv2 commented:
Please ignore...

cortner commented:
@casv2 the dictionaries were supposed to go into ACE1pack, which I've started. I wouldn't mess around too much with IPFitting and put all the effort of fixing the interface into ACEfit please.

cortner commented:
I'll try to produce an example how I'm envisioning how ACE1pack will work and let you know immediately.

casv2 commented:
Ok, not much has changed actually, a simple fit_dict in which we put the params. Before we'd have `(:lsqr, (1e-3, 1e-6))` which is not user friendly at all. That's now 

`fit_dict = Dict("solver" => :lsqr,
                          "tol" => 1e-3,
                          "damp"  => 1e-6)`

Which is much nicer, apart from that nothing has really changed. Also removed the ACE dependency whereby one can now add a `P` matrix to the `fit_dict`do the Laplacian regularisation. I feel this makes more sense to be honest. Also fixed the `asmerrs=true`.

This can actually be merged I just checked. I tried a clean install but for some reason I was on /cortner/IPFitting.jl#dev instead of on here. Checking out this ACEsuit/IPFitting#dev now works perfectly well.

cortner commented:
entire up to you really, I just want to discourage doing too much extra work other than cleanup.

casv2 commented:
The work is already done, and I think this interface gives you a lot more information as to what you're doing.
 
I'll make sure it works and write some clear tutorial and how to use it with LAMMPs. This removes the ACE dependency and adds an extra line when trying to do Laplacian regularisation but apart from that I think it's much more clear. We now have keywords we have to specify `tol`, `damp` which we can now write documentation around. Before it was just completely unclear what you were doing.

I'll finish writing the tutorials, confirm again that this latest /dev branch makes everything work and then get back to you if we want to merge. Then tomorrow I'll ask others to see whether they can get it to work on their machines too.
